### PR TITLE
Visitor can Register for an account

### PR DESCRIPTION
**Pull request by vick3d**
### User Story
As a Visitor,
In order to access unique features on the news app,
I would like to be able to register for an account on the site.

### Tasks
Added devise gem
Add link from index to sign up page
Styled sign-up page
re-direction to index page after signing up

### Screenshot
image

### CODE REVIEW

**Comment by vick3d** [comment]
We have generated the Devise views and will as we proceed with the development re-write and style the needed ones to HAML.  (like we have done with the sign up page) Would you rather prefer that we delete the ones that are not used at the moment?

File: app/views/devise/confirmations/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<h2>Resend confirmation instructions</h2>
```
Comment (oliverochman) [inline app/views/devise/confirmations/new.html.erb]: remove the ones that you don't modify and add them when you need them

Comment (vick3d) [reply app/views/devise/confirmations/new.html.erb]: done

File: config/initializers/devise.rb [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+  # note that it will be overwritten if you use your own mailer class
+  # with default "from" parameter.
+  config.mailer_sender = 'please-change-me-at-config-initializers-devise@example.com'
+
+  # Configure the class responsible to send e-mails.
+  # config.mailer = 'Devise::Mailer'
+
+  # Configure the parent class responsible to send e-mails.
+  # config.parent_mailer = 'ActionMailer::Base'
+
```
Comment (oliverochman) [inline config/initializers/devise.rb]: please clean this file up

File: features/Visitor_can_register_for_an_account.feature [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+   As a Visitor,
+   In order to access unique features on the news app,
+   I would like to be able to register for an account on the site.
+
+   Scenario:
+      Given I visit the "landing" page
+      And I click on "Become a subscriber"
+      And I fill in "Email" with "person@mail.com"
+      And I fill in "Password" with "12345678"
```
Comment (oliverochman) [inline features/Visitor_can_register_for_an_account.feature]: make the password something readable like "password"

File: app/views/devise/mailer/password_change.html.erb [hunk 4]
```diff
@@ -0,0 +1,3 @@
+<p>Hello <%= @resource.email %>!</p>
+
```
Comment (LauraRe) [inline app/views/devise/mailer/password_change.html.erb]: remove extra space

File: app/views/devise/mailer/reset_password_instructions.html.erb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<p>Hello <%= @resource.email %>!</p>
+
```
Comment (LauraRe) [inline app/views/devise/mailer/reset_password_instructions.html.erb]: remove space

File: app/views/devise/mailer/reset_password_instructions.html.erb [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+<p>Someone has requested a link to change your password. You can do this through the link below.</p>
+
+<p><%= link_to 'Change my password', edit_password_url(@resource, reset_password_token: @token) %></p>
+
```
Comment (LauraRe) [inline app/views/devise/mailer/reset_password_instructions.html.erb]: remove spaces line 4 and 6

File: app/views/devise/mailer/unlock_instructions.html.erb [hunk 7]
```diff
@@ -0,0 +1,7 @@
+<p>Hello <%= @resource.email %>!</p>
+
+<p>Your account has been locked due to an excessive number of unsuccessful sign in attempts.</p>
+
+<p>Click the link below to unlock your account:</p>
+
```
Comment (LauraRe) [inline app/views/devise/mailer/unlock_instructions.html.erb]: remove spaces line 2, 4 and 6

File: app/views/devise/registrations/new.html.haml [hunk 8]
```diff
@@ -0,0 +1,25 @@
+
```
Comment (LauraRe) [inline app/views/devise/registrations/new.html.haml]: remove space

File: config/initializers/devise.rb [hunk 9]
```diff
@@ -0,0 +1,299 @@
...
+  #
+  # ActiveSupport.on_load(:devise_failure_app) do
+  #   include Turbolinks::Controller
+  # end
+
+  # ==> Configuration for :registerable
+
+  # When set to false, does not sign a user in automatically after their password is
+  # changed. Defaults to true, so a user is signed in automatically after changing a password.
+  # config.sign_in_after_change_password = true
```
Comment (LauraRe) [inline config/initializers/devise.rb]: should we delete some comments?

File: features/Visitor_can_register_for_an_account.feature (hunk 3 above)
Comment (vick3d) [reply features/Visitor_can_register_for_an_account.feature]: done

File: config/initializers/devise.rb (hunk 2 above)
Comment (vick3d) [reply config/initializers/devise.rb]: and done

File: app/models/user.rb [hunk 10]
```diff
@@ -0,0 +1,5 @@
+class User < ApplicationRecord
+:omniauthable
```
Comment (oliverochman) [inline app/models/user.rb]: you dont need this, please remove it

Comment (amandagibson) [reply app/models/user.rb]: done

**Pull request merged**